pkg/bootstrap.go

```diff
@@ -0,0 +1,103 @@
+
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/doguregistry"
+	kubernetes2 "github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/kubernetes"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/reconciler"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/serializer"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/serializer/blueprintMaskV1"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/serializer/blueprintV2"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/application"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/config"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+)
+
+// ApplicationContext contains vital application parts for this operator.
+type ApplicationContext struct {
+	RemoteDoguRegistry         domainservice.RemoteDoguRegistry
+	BlueprintSpecRepository    domainservice.BlueprintSpecRepository
+	BlueprintSpecDomainUseCase *domainservice.ValidateDependenciesDomainUseCase
+	DoguInstallationUseCase    *application.DoguInstallationUseCase
+	BlueprintSpecUseCase       *application.BlueprintSpecUseCase
+	BlueprintSerializer        serializer.BlueprintSerializer
+	BlueprintMaskSerializer    serializer.BlueprintMaskSerializer
+	Reconciler                 *reconciler.BlueprintReconciler
+}
+
+// Bootstrap creates the ApplicationContext.
+func Bootstrap(restConfig *rest.Config, eventRecorder record.EventRecorder, namespace string) (*ApplicationContext, error) {
+	blueprintSerializer := blueprintV2.Serializer{}
+	blueprintMaskSerializer := blueprintMaskV1.Serializer{}
+	ecosystemClientSet, err := createEcosystemClientSet(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	blueprintSpecRepository := kubernetes2.NewBlueprintSpecRepository(
+		ecosystemClientSet.EcosystemV1Alpha1().Blueprints(namespace),
+		blueprintSerializer,
+		blueprintMaskSerializer,
+		eventRecorder,
+	)
+
+	remoteDoguRegistry, err := createRemoteDoguRegistry(err)
+	if err != nil {
+		return nil, err
+	}
+
+	blueprintSpecDomainUseCase := domainservice.NewValidateDependenciesDomainUseCase(remoteDoguRegistry)
+	doguInstallationUseCase := &application.DoguInstallationUseCase{}
+	blueprintUseCase := application.NewBlueprintSpecUseCase(blueprintSpecRepository, blueprintSpecDomainUseCase, doguInstallationUseCase)
+	blueprintReconciler := reconciler.NewBlueprintReconciler(blueprintUseCase)
+
+	return &ApplicationContext{
+		RemoteDoguRegistry:         remoteDoguRegistry,
+		BlueprintSpecRepository:    blueprintSpecRepository,
+		BlueprintSpecDomainUseCase: blueprintSpecDomainUseCase,
+		DoguInstallationUseCase:    doguInstallationUseCase,
+		BlueprintSpecUseCase:       blueprintUseCase,
+		BlueprintSerializer:        blueprintSerializer,
+		BlueprintMaskSerializer:    blueprintMaskSerializer,
+		Reconciler:                 blueprintReconciler,
+	}, nil
+}
+
+func createRemoteDoguRegistry(err error) (*doguregistry.Remote, error) {
+	remoteConfig, err := config.GetRemoteConfiguration()
```

Comment:
🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4009(related information): assignment to err